lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
```

Comment:
Look at this first: https://github.com/Shopify/semian/pull/54#discussion-diff-42760663

I don't think we've decided yet what to do with it in this case of AtomicInteger, its part of `SharedMemoryObject#execute_atomically`